src/routes/mod_creation.rs

```diff
@@ -110,25 +104,33 @@ struct ModCreateData {
 #[post("api/v1/mod")]
 pub async fn mod_create(
     mut payload: Multipart,
-    client: Data<Client>,
+    client: Data<PgPool>,
```

Comment:
We need to make a transaction here, and add error handling at each place that this could return an error to delete files that have been uploaded when the transaction fails.